src/DebugBar/DebugBar.php

```diff
@@ -0,0 +1,121 @@
+<?php
+/**
+ * @author    Andrew Coulton <andrew@ingenerator.com>
+ * @licence   proprietary
+ */
+
+namespace Ingenerator\KohanaExtras\DebugBar;
+
+
+use DebugBar\Bridge\DoctrineCollector;
+use DebugBar\StandardDebugBar;
+use Doctrine\ORM\EntityManagerInterface;
+use Ingenerator\KohanaExtras\DebugBar\DebugLogWriter;
+use Ingenerator\KohanaExtras\DebugBar\DebugStackTrace;
+
+/**
+ * Simple wrapper for DebugBar to manage availability on the site
+ *
+ * @package Ingenerator\Debug
+ */
+class DebugBar
+{
+    /**
+     * @var \DebugBar\DebugBar
+     */
+    protected $bar;
+
+    /**
+     * @var bool
+     */
+    protected $is_active = FALSE;
+
+    /**
+     * @var \Dependency_Container
+     */
+    protected $dependencies;
+
+    /**
+     * Explicitly and intentionally takes no direct constructor dependencies to allow
+     * safe access to the debug bar from anywhere in the system even in production with
+     * no impact unless the bar has been initialised.
+     *
+     * @param \Dependencies $dependencies
+     */
+    public function __construct(\Dependencies $dependencies)
+    {
+        $this->dependencies = $dependencies;
+    }
+
+    public function initialise()
+    {
+        if (\Kohana::$environment !== \Kohana::DEVELOPMENT) {
+            throw new \BadMethodCallException(
+                'Cannot use debug bar outside development environment'
+            );
+        }
+        $this->bar = new StandardDebugBar;
+        $this->initDoctrineBar($this->dependencies->get('doctrine.entity_manager'));
+        $this->initLogCollector($this->dependencies->get('kohana.log'));
+        $this->is_active = TRUE;
+    }
+
+    protected function initDoctrineBar(EntityManagerInterface $em)
+    {
+        $stack = new DebugDoctrineQueryTrace;
+        $em->getConnection()->getConfiguration()->setSQLLogger($stack);
+        $this->bar->addCollector(new DoctrineCollector($stack));
+        $this->bar->getJavascriptRenderer()->addAssets(
```

Comment:
@craig410 here's how I've configured it to build those styles directly into the debug bar CSS we compile below. The Collector classes can also implement the `AssetProvider` interface and return an array of the assets they need, so if we're ever adding a new panel in future that's also an option. Can either reference an external file, or add things as inline strings in which case they just get added into the concated file.